app/src/app/user/profile/profile.ts

```diff
@@ -24,24 +24,35 @@ namespace app.user.profile {
                     onBoard: false
                 },
                 resolve: /*@ngInject*/{
-                    emailConfirmationToken:(
+                    emailConfirmed:(
                         userService:common.services.user.UserService,
                         $stateParams:IStateParams,
                         ngJwtAuthService:NgJwtAuth.NgJwtAuthService,
                         notificationService:common.services.notification.NotificationService,
-                        $location:ng.ILocationService
+                        $location:ng.ILocationService,
+                        $q:ng.IQService
                     ) => {
                         if(!_.isEmpty($stateParams.emailConfirmationToken)) {
-                            userService.confirmEmail(<common.models.User>ngJwtAuthService.getUser(), $stateParams.emailConfirmationToken)
+
+                            let emailToken = $stateParams.emailConfirmationToken;
+
+                            return userService.confirmEmail(<common.models.User>ngJwtAuthService.getUser(), emailToken)
                                 .then(() => {
-                                    notificationService.toast('Your email has successfully been updated').pop();
 
-                                    $location.search('emailConfirmationToken', null);
```

Comment:
By default it seems that when you change the URL params a state change is initiated which means the profile controller is loaded again. This is what causes the multiple state changes when confirming an email (one state change for removing loginToken and one more for removing emailConfirmToken). If you leave the emailConfirmationToken and loginToken in the URL only one initial state change occurs.

See http://stackoverflow.com/questions/20884551/set-url-query-parameters-without-state-change-using-angular-ui-router

As suggested in Stackoverflow we can set reload to false but we would have to do this globally as well, I'm not sure if this is a good idea or how it will affect things later on. What do you think?

> reloadOnSearch v0.2.5: Boolean (default true). If false will not retrigger the same state just because a search/query parameter has changed. Useful for when you'd like to modify $location.search() without triggering a reload.